internal/status_test.go

```diff
@@ -0,0 +1,14 @@
+package internal
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestComposeStatusString(t *testing.T) {
+	line := composeStatusString(0, 1000, 0, 1, true, []string{"-", "\\", "|", "/", "-"}, 0, time.Now(), true, 20)
+	expected := "\r-[ ]          0/1 Complete          0B / 1,000B          0s Elapsed"
+	assert.Equal(t, expected, line)
+}
```

Comment:
Let's add the following tests:
- Test the happy, simple path: just 2 small resources, test the initialization of the StatusLine, test that the status string looks okay at first, increment, test again the string, re-increment, test the string, then test the end state of the status string.
- Test the massive case: test the status string when there are 1000 resources in the file
- Test the error cases
  - Can't find the size
  - Passed context gets cancelled
  - other error cases?

Reply:
How would you recommend I go about generating 1000 test resources?

Reply:
Something like this:
```
	content := `abcdef`
	port := test.TestHttpHandler(content, t)
	resources := []Resource{}
	algo := "sha256"
	for i := 0; i < 1000; i++ {
		resource := Resource{Urls: []string{fmt.Sprintf("http://localhost:%d/test%d.html", port, i)}, Integrity: fmt.Sprintf("%s-vvV+x/U6bUC+tkCngKY5yDvCmsipgW8fxsXG3Nk8RyE=", algo), Tags: []string{}, Filename: ""}
		resources = append(resources, resource)
	}
	assert.Len(t, resources, 1000)
```